tools/build/Library/Version.php

```diff
@@ -74,10 +74,11 @@ public function __construct($version)
     {
         $this->version = $version;
         $versions = explode('.', $version);
-        $this->majorVersionString = $versions[0] . '.' . $versions[1];
-        $this->majorVersion = (int) ($versions[0] . $versions[1]);
-        $this->minorVersion = (int) $versions[2];
-        $this->releaseVersion = (int) $versions[3];
+
+        $this->majorVersionString = $versions[0];
+        $this->majorVersion = (int) ($versions[0]);
+        $this->minorVersion = (int) $versions[1];
```

Comment:
Will this work for both "legacy versions" 1.7.8.0 and "modern versions" 8.0.0 ?

Reply:
This will work only for modern versions I think

Reply:
Yes it's working only for newer version as this tool is not designed to work on previous version :)

Reply:
Isn't this class already fixed in the core? Why not copy that one over?

Otherwise this class should be updated to erase all references to the old format.

Reply:
The PR aims to fix the CI problem first, we will see later for improvements